### PR TITLE
Added simple mention handling

### PR DESCRIPTION
**Pull request by TeddyDesTodes**
Added handling of @Mention handling

`@Name` will be expanded to `@<Name http://url/to/twtxt.txt>` if you follow "Name" Unknown Names will be ignored.

Nothing was changed on the timeline rendering, `@<Name http://url/to/twtxt.txt>` will be transformed to `@Name` before printing. No benefits for the user for now, but something like "show the URL if you don't follow this user already" could be implemented.

I'm not really fond of the get_source_by_* functions due to the brute-force nature of them, but I do not really want to change the way the Sources are handled internall

### CODE REVIEW

File: twtxt/helper.py [hunk 1]
```diff
@@ -85,3 +87,53 @@ def sort_and_truncate_tweets(tweets, direction, limit):
...
+        source = get_source_by_name(match.group(1))
+        if source is None:
+            return '@' + match.group(1)
+        return format.format(name=source.nick,
+                             url=source.url)
+
+    return short_mention_re.sub(handle_mention, text)
+
+
+def get_source_by_url(url):
```
Comment (timofurrer) [inline twtxt/helper.py]: Use list comprehension ..

File: twtxt/helper.py [hunk 2]
```diff
@@ -85,3 +87,53 @@ def sort_and_truncate_tweets(tweets, direction, limit):
...
+    sources = click.get_current_context().obj["conf"].following
+    for source in sources:
+        if source.url == url:
+            return source
+    return None
+
+
+def get_source_by_name(name):
+    sources = click.get_current_context().obj["conf"].following
+    for source in sources:
```
Comment (timofurrer) [inline twtxt/helper.py]: Use list comprehension ...

**Comment by timofurrer** [comment]
What about highlight mention tags in the timeline?

**Comment by TeddyDesTodes** [comment]
as for highlighting:
if is possible to do so, but the truncating to 140 chars happens after formatting, and i don't believe it will be handled in a predictable way.

just replace line 131 in helper.py with:
```
        if name == 'teddydestodes':
            fg = 'magenta'
        return click.style('@' + name, bold=True, fg=fg)
```

**Comment by buckket** [comment]
Merged into mentions branch and updated slightly, discussion is over at #6.

**Pull request closed without merging**